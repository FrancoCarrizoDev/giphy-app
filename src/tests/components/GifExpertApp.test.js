import { shallow } from "enzyme"
import {GifExpertApp} from '../../GifExpertApp'
import React from 'react';
import '@testing-library/jest-dom'

describe('Test in component', () => {
    test('should show correctly', () => {
        
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('should show correctly an list category', () => {
        
        const category = ['One Punch', 'Goku']
        const wrapper = shallow(<GifExpertApp defaultCategories={category}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(category.length)
    })

    
    
})
