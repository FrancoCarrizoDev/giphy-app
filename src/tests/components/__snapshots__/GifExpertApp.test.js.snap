// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in component should show correctly an list category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
